Bluepill w/ Xcode 11 support, more logging, minor fixes and partial revert of bugfix in PR#338

ravimandala opened the pull request:
I am looking for a Bluepill version to be used w/ Xcode 11.
Can you please merge this PR and generate a Bluepill release with the fix to https://github.com/linkedin/bluepill/pull/338 reverted and more logging added (cherry-picked a commit from your branch https://github.com/linkedin/bluepill/tree/more-logging)? I tested this on machines with Xcode 11 GM Seed#1.
Thanks. \cc @ob

ob commented on BPSampleApp/BPSampleAppCrashingTests/BPSampleAppCrashingTests.m:
```diff
@@ -15,21 +15,6 @@ @interface BPSampleAppCrashingTests : XCTestCase
 
 @implementation BPSampleAppCrashingTests
 
-NSString *attemptFromSimulatorVersionInfo(NSString *simulatorVersionInfo) {
```
Let's keep this and the tests, just disable them.

ravimandala replied:
Done.

ob commented on bp/src/Bluepill.m:
```diff
@@ -600,7 +603,8 @@ - (void)finishWithContext:(BPExecutionContext *)context {
         [BPUtils printInfo:INFO withString:@"No more tests to run."];
-        self.finalExitStatus = context.exitStatus;
+        [BPUtils printInfo:INFO withString:@"%s:%d finalExitStatus = %@", __FILE__, __LINE__, [BPExitStatusHelper stringFromExitStatus:self.finalExitStatus]];
+        self.finalExitStatus = context.finalExitStatus | context.exitStatus;
```
Maybe add a comment that #338 was reverted while we investigate other problems.

ravimandala replied:
Done.

ravimandala replied:
Done.

ob commented on bp/tests/BluepillTests.m:
```diff
@@ -232,22 +232,6 @@ - (void)testReportWithAppCrashingAndRetryOnlyFailedTestsSet {
     [self assertGotReport:junitReportPath isEqualToWantReport:expectedFilePath];
 }
 
-- (void)testAppCrashingAndRetryReportsCorrectExitCode {
```
Please keep the test, just disabled.

ravimandala replied:
Done.

ravimandala commented:
@ob I updated this PR with the below changes...

- Picked changes from more-logging branch;
- Made minor changes to fix some warnings and fixed the bluepill.sh for missing `xcpretty` use case;
- Partial revert of "Fix wrong exit status (#338)"

Can you please take another look?
